ruma-common/src/push.rs

```diff
@@ -81,6 +81,20 @@ impl Ruleset {
     }
 }
 
+impl IntoIterator for Ruleset {
+    type Item = AnyPushRule;
+    type IntoIter = std::vec::IntoIter<Self::Item>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.content.into_iter().map(Into::into)
+            .chain(self.override_.into_iter().map(Into::into))
+            .chain(self.room.into_iter().map(Into::into))
+            .chain(self.sender.into_iter().map(Into::into))
+            .chain(self.underride.into_iter().map(Into::into))
+            .collect::<Vec<_>>().into_iter()
```

Comment:
If we put a bit more work into this, it can work without allocating. But we can also defer that to later.

Reply:
I will try something.